File: review-agent/app/schemas/report.py
from datetime import date
from typing import Literal

from pydantic import BaseModel, Field


class DateRange(BaseModel):
    start: date
    end: date


class AnalyzeRequest(BaseModel):
    product_id: str = Field(min_length=1)
    date_range: DateRange | None = None


class AnalyzeResponse(BaseModel):
    task_id: str


class TaskStatusResponse(BaseModel):
    task_id: str
    status: Literal["pending", "running", "completed", "failed"]


class SimilarCase(BaseModel):
    review_id: str
    content: str
    similarity_score: float


class Summary(BaseModel):
    total_count: int
    negative_count: int
    negative_rate: float
    avg_sentiment: float


class SentimentTrendItem(BaseModel):
    month: str
    avg_sentiment: float


class ReportResponse(BaseModel):
    summary: Summary
    reason_categories: dict[str, int]
    keywords: list[str]
    sentiment_trend: list[SentimentTrendItem]
    suggestions: list[str]
    similar_cases: list[SimilarCase]

File: review-agent/frontend/src/types/report.ts
export type TaskStatus = "pending" | "running" | "completed" | "failed"

export interface AnalyzeRequest {
  product_id: string
  date_range?: {
    start: string
    end: string
  }
}

export interface AnalyzeResponse {
  task_id: string
}

export interface TaskStatusResponse {
  task_id: string
  status: TaskStatus
}

export interface Summary {
  total_count: number
  negative_count: number
  negative_rate: number
  avg_sentiment: number
}

export interface SentimentTrendItem {
  month: string
  avg_sentiment: number
}

export interface SimilarCase {
  review_id: string
  content: string
  similarity_score: number
}

export interface ReportResponse {
  summary: Summary
  reason_categories: Record<string, number>
  keywords: string[]
  sentiment_trend: SentimentTrendItem[]
  suggestions: string[]
  similar_cases: SimilarCase[]
}
